comment/models/comments.py

```diff
@@ -38,6 +43,17 @@ def replies(self, include_flagged=False):
             return self.__class__.objects.filter(parent=self).order_by('posted')
         return self.__class__.objects.all_exclude_flagged().filter(parent=self).order_by('posted')
 
+    def _set_unique_urlhash(self):
+        if not self.urlhash:
+            hash_val = self.__class__.objects.generate_urlhash()
+            self.urlhash = hash_val
+            while self.__class__.objects.filter(urlhash=hash_val).exists():
+                self.urlhash = self.__class__.objects.generate_urlhash()
```

Comment:
This is always evaluated to True in case urlhash exists.
```suggestion
    def _set_unique_urlhash(self):
        if not self.urlhash:
            hash_val = self.__class__.objects.generate_urlhash()
            self.urlhash = hash_val
            while self.__class__.objects.filter(urlhash=self.hash_val).exists():
                self.urlhash = self.__class__.objects.generate_urlhash()
```

Reply:
Both `self.urlhash` and `hash_val` have the same values. Also, if it were always true, won't the loop run indefinitenly?

Do I miss something here?

Reply:
No thery are not, `hash_val` is const and doesn't change inside while loop, `self. urlhash` is updating inside while loop. therefore you have to test `self.urlhash` if exist. to be honest `hash_val` is redundant and has no meaning, yo could do `self.urlhash = self.__class__.objects.generate_urlhash()` directly.

Reply:
Yes, I got you. A blunder on my side. Thanks for pointing this out.

Reply:
It's not a big deal, we all do this frequently

Reply:
I will try to add the test for migrations now and push this in a bit.

Reply:
And remember whatever happens, I love you for solving this mock thing :-p This was really annoying me for long.

Reply:
I appreciate your efforts and your contributions are really valuable, I learned a lot from you as well. Many thanks!